platform/app/src/routes/WorkList/WorkList.tsx

```diff
@@ -197,6 +200,7 @@ function WorkList({
   useEffect(() => {
     const fetchSeries = async studyInstanceUid => {
       try {
+        await dataSource.initialize({ params, query });
```

Comment:
This doesn't resonate with me, we are initializing the data source 3 times after this PR (used to be one time in the mode.tsx). 

Should we call it something else?
Should we skip it if already initialized?

Reply:
Good call. I added a logic to skip if its already initialized. I added this code here so we want initialization before loading things.

Reply:
I too am concerned with having multiple initialize calls. One thing that concerns be the most is that prior to this PR, one could use some of the methods without calling `initialize` first. In particular, `getConfig` is one such method and I regret to let you know that now upload is broken with this PR. So in the future I can see more "places" where initialize has to be called. Is there a more central/early location to call initialize? I feel bad suggesting there is a problem without an actual suggestion of how to fix it, but I am open to discussing it. Thoughts?

image.png

Reply:
> I too am concerned with having multiple initialize calls. One thing that concerns be the most is that prior to this PR, one could use some of the methods without calling `initialize` first. In particular, `getConfig` is one such method and I regret to let you know that now upload is broken with this PR. So in the future I can see more "places" where initialize has to be called. Is there a more central/early location to call initialize? I feel bad suggesting there is a problem without an actual suggestion of how to fix it, but I am open to discussing it. Thoughts?

Maybe we should place the initialize somewhere else. Currently it doest work like I would expect, if I go to the worklist, another code from the dicomweb datasource executes before initialize is called. I'll revisit the PR and see where I place it so we don't have to call it twice.

Reply:
![image](https://github.com/OHIF/Viewers/assets/109477394/fae3caec-c648-47c7-8224-4da2c1fc54d9)